### PR TITLE
chore(wdio): port lifecycle-async tests from karma to wdio

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
Porting the `lifecycle-async` test suite from Karma to Wdio!

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

Nothing too crazy going on! Some comments pointing out a few things though.

### CODE REVIEW

File: test/wdio/lifecycle-async/cmp-a.test.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  beforeEach(() => {
+    render({
+      template: () => <lifecycle-async-a></lifecycle-async-a>,
+    });
+  });
+
+  it('fire load methods in order', async () => {
+    await $('lifecycle-async-a').waitForExist();
+
+    await browser.pause(500);
```
Comment (alicewriteswrongs) [inline test/wdio/lifecycle-async/cmp-a.test.tsx:14]: unfortunately without these `borwser.pause` calls here things aren't in place by the time we do all the `toHaveText` checks

**Review by alicewriteswrongs — COMMENTED** [review]
just pointing out one thing

File: test/wdio/lifecycle-async/util.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export function timeout(ms: number): Promise<void> {
```
Comment (christian-bromann) [inline test/wdio/lifecycle-async/util.ts:1]: Do we need this?

File: test/wdio/lifecycle-async/cmp-a.test.tsx (hunk 1 above)
Comment (christian-bromann) [inline test/wdio/lifecycle-async/cmp-a.test.tsx:14]: Would this work?
```suggestion
    await $('lifecycle-async-a li').waitForExist();
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]

File: test/wdio/lifecycle-async/util.ts (hunk 2 above)
Comment (alicewriteswrongs) [reply test/wdio/lifecycle-async/util.ts:1]: yeah two of the components use it, it's not used in the test file but in `cmp-b` and `cmp-c`

Comment (christian-bromann) [reply test/wdio/lifecycle-async/util.ts:1]: ah, ok didn't recognize that. I thought it was a way to "pause" the test. All good then.

File: test/wdio/lifecycle-async/cmp-a.test.tsx (hunk 1 above)
Comment (alicewriteswrongs) [reply test/wdio/lifecycle-async/cmp-a.test.tsx:14]: I do still need the pause here, if I just do the `await $('lifecycle-async-a li')` then the tests ends up being flaky and seems to faily around 50% of the time

Comment (alicewriteswrongs) [reply test/wdio/lifecycle-async/cmp-a.test.tsx:14]: It looks like what can happen is that it stops waiting when only one `li` is on the DOM, so the check for `.length` doesn't work :/

**Review by rwaskiewicz — APPROVED** [review]

File: test/wdio/lifecycle-async/cmp-a.test.tsx (hunk 1 above)
Comment (christian-bromann) [reply test/wdio/lifecycle-async/cmp-a.test.tsx:14]: I raised an issue for this and will take a look eventually: https://github.com/webdriverio/expect-webdriverio/issues/1493

**Review by christian-bromann — APPROVED** [review]
👍

**Pull request merged**